docs/modules/ROOT/pages/explanation/decisions/portal-api.adoc

```diff
@@ -0,0 +1,78 @@
+= {product} Portal API
+
+== Problem
+
+For the xref:explanation/system/details-ui.adoc[{product} Portal] an API endpoint is needed which implements the backend functionality.
+This API endpoint is described in xref:explanation/system/details-external-services.adoc[External Services].
+As documented, this API endpoint connects various third-party backend systems together and presents it in a well-defined uniform way for the Portal UI to be consumed.
+It must be possible to connect various different backend systems which implement the same functionality so that it is a pluggable system and can be re-used by other parties, which might have different backends for the same task.
+{product} API only specifies the interface, the implementation is then user specific.
+
+== Relevant requirements
+
+* xref:references/quality-requirements/maintainability/portal-api-modularity.adoc[Portal API Modularity]
+* xref:references/quality-requirements/reliability/portal-api-data-recoverability.adoc[Portal API Data Recoverability]
+* xref:references/quality-requirements/performance/portal-api-data-freshness.adoc[Portal API Data Freshness]
+
+== Proposals
+
+[qanda]
+Custom API implementation::
+Implementation of a custom API which connects all the backend system synchronously and provides the abstraction to them.
+Authentication, authorization and other core API features will have to be implemented specific to this API.
+
+Kubernetes API::
+Reuse the Kubernetes API to leverage all the plumbing which already exists: Authentication, RBAC, scalability, well-defined resource-model, extensibility with `CRDs` and Aggregated API server.
+This seems to be a common topic in the Cloud Native world, see for example the experiment called https://github.com/kcp-dev/kcp[kcp - a minimal Kubernetes API server].
+- Async vs. Sync.
+- The Crossplane model
+- Backend system as status
+
+== Decision
+
+Kubernetes API
+
+== Rationale
+
+The Kubernetes API server is a perfect match for this use-case. With it, we can many core API plumbing features "for free":
+
+Authentication::
+We can leverage what we already use for the {zone}s.
+Authentication is built-in to the Kubernetes API server and it's even pluggable.
+
+Authorization::
+We can make use of standard Kubernetes RBAC access control and could even enhance it with a policy engine like Kyverno.
+
+Asynchronous::
+It allows us to build upon an asynchronous architecture which when working with multiple backend systems will prove to be more stable and easier to implement.
+
+Persistent Storage::
+Storing data is built-in directly in the core.
+We can store temporary data without having to implement a storage backend.
+
+API Extensibility::
+The API can easily be extended by leveraging `CustomResourceDefinition` (CRDs).
+
+Scalability and Availability::
+It's a core feature of the Kubernetes API server to be highly scalable.
+High-availability possibilities are directly built-in.
```

Comment:
This is true for both proposals, since with Ingress we can also make REST API HA. So I would remove it since it has no impact. Unless there's an important aspect in how this HA is achieved, but I don't see one in either case.

Reply:
Removed

Reply:
Removed